pkg/reconciler/broker/broker.go

```diff
@@ -121,6 +121,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 		logging.FromContext(ctx).Debug("Broker reconciled")
 	}
 
+	// Since the reconciler took a crack at this, make sure it's reflected
+	// in the status correctly.
+	broker.Status.ObservedGeneration = original.Generation
```

Comment:
Just curious why you decided to set that particular field outside of `updateStatus()`? Nothing inside `reconcile()` seem to alter the object's `Generation`, so wouldn't it be possible to start with `obj.Status.ObservedGeneration = obj.Generation` _inside_ every `updateStatus()` func?

Reply:
Solid point. Thinking is that we eventually want to hoist these kinds of "mundane" tasks away from each "little r" and into the "big R", as per:
https://docs.google.com/document/d/1tMZ5-uZHaBeV_eF13zw_J-PDh6jtd10vNhKabD4SBrI/edit#

Idea being that the less custom logic the "little r" has to do, the better off we are. Does that make sense?

Reply:
Yup, thanks for explaining the reasoning!